### PR TITLE
Update flow

### PR DESCRIPTION
**Pull request by duailibe**
In order to support JSX fragments (#3237) we need to update flow to version 0.59. Since there were some relevant changes I decided to do another sync of the tests.

### CODE REVIEW

**Comment by duailibe** [comment]
cc @lydell

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/node_modules
+node_modules
```
Comment (duailibe) [inline .gitignore]: Some tests have a `node_modules` folder to test module resolution but `jest` doesn't run them and I'm not sure they would bring relevant information to tests anyway.

File: tests/flow_class_field/jsfmt.spec.js [hunk 2]
```diff
@@ -1 +1 @@
-run_spec(__dirname, null, ["babylon"]);
+run_spec(__dirname, { parser: "babylon" });
```
Comment (duailibe) [inline tests/flow_class_field/jsfmt.spec.js]: It's related to #1481 but now this code only parses in babylon and not flow anymore. Maybe the babylon fix should be reverted? cc @existentialism

File: tests/newline/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -3,13 +3,15 @@
...
 `;
 
 exports[`backslash_2029.js 1`] = `
 1;/*a*///b
/*c*/2
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-1; /*a*/ //b
/*c*/2
+1; /*a*/ //b
+/*c*/ 2;
```
Comment (duailibe) [inline tests/newline/__snapshots__/jsfmt.spec.js.snap]: Related to #883

not sure what to do here

**Comment by lydell** [comment]
Regarding syncing the Flow tests: See #2336. Do any tests contain new syntax or something that we'd like to copy? Otherwise it might not be worth syncing. Perhaps we should even remove the sync script.

**Comment by duailibe** [comment]
I don't think they have any new syntax, they've only removed stuff. Maybe we should only update the existing ones?

**Comment by duailibe** [comment]
I think something's changed on the flow-parser, now the dist tests are failing with:

`Dynamic requires are not currently supported by rollup-plugin-commonjs`

**Comment by duailibe** [comment]
I managed to add a hack to work around flow-parser’s dynamic require. Hopefully tests will pass now!

File: scripts/build/rollup.parser.config.js [hunk 4]
```diff
@@ -18,6 +18,15 @@ export default Object.assign(baseConfig, {
...
       : {},
+    // In flow-parser 0.59.0 there's a dynamic require: `require(s8)` which not
+    // supported by rollup-plugin-commonjs, so we have to replace the variable
+    // by its value before bundling.
+    parser === "flow"
+      ? replace({
+          "require(s8)": 'require("fs")',
+          include: "node_modules/flow-parser/flow_parser.js"
+        })
+      : {},
```
Comment (vjeux) [inline scripts/build/rollup.parser.config.js]: cc @mroch, not sure it was intentional but would be great to remove dynamic requires

File: tests/flow/fetch/headers.js [hunk 5]
```diff
@@ -2,15 +2,15 @@
 
 const a = new Headers("'Content-Type': 'image/jpeg'"); // not correct
 const b = new Headers(['Content-Type', 'image/jpeg']); // not correct
-const c = new Headers({'Content-Type', 'image/jpeg'}); // correct
+const c = new Headers({'Content-Type': 'image/jpeg'}); // correct
```
Comment (vjeux) [inline tests/flow/fetch/headers.js]: How was it even working!?

Comment (mroch) [reply tests/flow/fetch/headers.js]: IIRC, it was valid short notation, and then not being typechecked properly

File: scripts/build/rollup.parser.config.js (hunk 4 above)
Comment (mroch) [reply scripts/build/rollup.parser.config.js]: this is js_of_ocaml optimizing things... and comes from its internal library. hrm.

also, `s8` is the random name it happened to assign to that hoisted string so this is definitely unstable across releases. will have to think about this...

Comment (duailibe) [reply scripts/build/rollup.parser.config.js]: Yeah, I'm aware this is a random name given in the ocaml -> JS build. But that shouldn't be a problem until we update flow again, and we don't do that often.

Comment (duailibe) [reply scripts/build/rollup.parser.config.js]: But it's funny that the `require("constants")` remained like that and `require("fs")` got hoisted.

**Pull request merged**